example/ipsec_crypto/odp_ipsec_stream.c

```diff
@@ -322,8 +322,8 @@ odp_packet_t create_ipv4_packet(stream_db_entry_t *stream,
 
 		encrypt_len = ESP_ENCODE_LEN(payload_len + sizeof(*esp_t),
 					     entry->esp.block_len);
-		memset(data, 0, encrypt_len - payload_len);
-		data += encrypt_len - payload_len;
+		for (int n = 0; n < encrypt_len - payload_len; n++)
+			*data++ = n + 1;
```

Comment:
While verifying this fix on our platform, observed that for tunnel mode, the total_len field of the inner IP is not set. It always comes as zero which seems incorrect.

Can we fix this as well with something like:
```
--- a/example/ipsec_crypto/odp_ipsec_stream.c                                    
+++ b/example/ipsec_crypto/odp_ipsec_stream.c                                    
@@ -272,6 +272,9 @@ odp_packet_t create_ipv4_packet(stream_db_entry_t *stream,   
                inner_ip = (odph_ipv4hdr_t *)data;                               
                memset((char *)inner_ip, 0, sizeof(*inner_ip));                  
                inner_ip->ver_ihl = 0x45;                                        
+               inner_ip->tot_len = odp_cpu_to_be_16(stream->length +            
+                                                    sizeof(*icmp) +             
+                                                    sizeof(*inner_ip));         
                inner_ip->proto = ODPH_IPPROTO_ICMPV4;                           
                inner_ip->id = odp_cpu_to_be_16(stream->id);                     
                inner_ip->ttl = 64;  ****
```

Reply:
Yes, let's fix that too. I submitted a new commit with a fix that is slightly (sizeof stream_pkt_hdr_t, to be exact) different from above. Please check if it is ok now.

Reply:
Yes. The fix looks good.

Reply:
> v2: added a new commit: example: ipsec: fix length field of inner IP header in tunnel mode

The PR looks good to me. You can add my reviewed-by if required.